baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/DeprecatedConjure.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.Matchers;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.MemberReferenceTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.Tree;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        severity = BugPattern.SeverityLevel.ERROR,
+        summary = "You should not use Conjure endpoints that are marked for removal as that may block"
+                + " upgrades in the near future. You can explicitly disable this check on a case-by-case basis using"
+                + " @SuppressWarnings(\"DeprecatedConjure\").")
+public final class DeprecatedConjure extends BugChecker
+        implements BugChecker.MethodInvocationTreeMatcher, BugChecker.MemberReferenceTreeMatcher {
+
+    private static final String CONJURE_CLIENT_ENDPOINT = "com.palantir.conjure.java.lib.internal.ClientEndpoint";
+
+    private static final Matcher<Tree> FULL_DEPRECATED_CONJURE_ENDPOINT_MATCHER =
+            Matchers.symbolMatcher((symbol, state) -> {
+                return ASTHelpers.hasAnnotation(symbol, CONJURE_CLIENT_ENDPOINT, state)
+                        && symbol.isDeprecatedForRemoval();
```

Comment:
I think we may want to reverse the order of this check for efficiency -- `ASTHelpers.hasAnnotation(symbol, CONJURE_CLIENT_ENDPOINT, state)` is relatively expensive and uncommon, where `symbol.isDeprecatedForRemoval()` should be a relatively simple bitwise comparison [`(flags_field & DEPRECATED_REMOVAL) != 0`](https://github.com/openjdk/jdk/blob/faa9c6909dda635eb008b9dada6e06fca47c17d6/src/jdk.compiler/share/classes/com/sun/tools/javac/code/Symbol.java#L378-L380).